src/Geno/C++/GUI/Widgets/TextEdit.cpp

```diff
@@ -977,6 +997,111 @@ void TextEdit::SetBoxSelection( File& rFile, int LineIndex, float XPosition )
 
 //////////////////////////////////////////////////////////////////////////
 
+void TextEdit::AddCursor( File& rFile, Cursor& rCursor )
+{
+	rCursor.Last = true;
+
+	for( int i = ( int )rFile.Cursors.size() - 1; i >= 0; i-- )
+	{
+		Cursor& rCur = rFile.Cursors[ i ];
+
+		if( rCursor.Position > rCur.Position )
+		{
+			rFile.Cursors.insert( rFile.Cursors.begin() + i + 1, rCursor );
+
+			for( int j = 0; j <= i; j++ )
+			{
+				rFile.Cursors[ j ].Last = false;
+			}
+
+			return;
+		}
+
+		rCur.Last = false;
+	}
+
+	rFile.Cursors.insert( rFile.Cursors.begin(), rCursor );
+
+} // AddCursor
+
+//////////////////////////////////////////////////////////////////////////
+
+TextEdit::Cursor& TextEdit::GetLastAddedCursor( File& rFile )
+{
+	GENO_ASSERT( rFile.Cursors.size() > 0 );
+
+	for( Cursor& rCursor : rFile.Cursors )
+	{
+		if( rCursor.Last )
+			return rCursor;
+	}
+
+	GENO_ASSERT( false );
+
+	// This should never happen
+	return rFile.Cursors[ 0 ];
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+TextEdit::Cursor& TextEdit::GetMainCursor( File& rFile )
+{
+	GENO_ASSERT( rFile.Cursors.size() > 0 );
+
+	for( Cursor& rCursor : rFile.Cursors )
+	{
+		if( rCursor.Main )
+			return rCursor;
+	}
+
+	GENO_ASSERT( false );
+
+	// This should never happen
+	return rFile.Cursors[ 0 ];
+
+} // GetMainCursor
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::EraseAllCursors( File& rFile, bool ExcludeMainCursor )
```

Comment:
What's a scenario where we would want to remove the main cursor?

Reply:
IDK, I think there was at one point but it's not used anymore, So I guess I could remove that parameter and possibly change the name of the function to say that the main cursor will remain.